### PR TITLE
Finished all the TODO in handler.go. 

### PR DESCRIPTION
**Pull request by xu-jiach**
Password validation requires the password to contain two among the four categories: Uppercase letter, Lowercase letter, Numbers, and special characters. 
Modified schema SQL to make the name unique.

### CODE REVIEW

File: backend/handler/handler.go [hunk 1]
```diff
@@ -253,6 +307,18 @@ func (h *Handler) AddItem(c echo.Context) error {
...
 	// http.StatusBadRequest(400)
+
+	// passing error if the image is bigger than 1MB
+	const MaxSize = 1 << 20 // 1MB
+	if file.Size > MaxSize {
+		return echo.NewHTTPError(http.StatusBadRequest, "file size exceeds limit")
+	}
+
+	// passing error if the image is not jpeg
+	if file.Header.Get("Content-Type") != "image/jpeg" {
```
Comment (noliver-ykc) [inline backend/handler/handler.go]: I would edit this to allow .png and .jpg as well since they are rather common photo types.

Comment (noliver-ykc) [reply backend/handler/handler.go]: `if file.Header.Get("Content-Type") != "image/jpeg" && file.Header.Get("Content-Type") != "image/png" && file.Header.Get("Content-Type") != "image/jpg" {`

File: backend/handler/handler.go [hunk 2]
```diff
@@ -291,15 +357,26 @@ func (h *Handler) Sell(c echo.Context) error {
...
 	// TODO: check req.UserID and item.UserID
 	// http.StatusPreconditionFailed(412)
+	if req.UserID != item.UserID {
+		return echo.NewHTTPError(http.StatusPreconditionFailed, "the item is not yours")
+	}
+
 	// TODO: only update when status is initial
 	// http.StatusPreconditionFailed(412)
+	if item.Status != domain.ItemStatusInitial {
+		return echo.NewHTTPError(http.StatusPreconditionFailed, "the item is in the state of initial. It has been on sale or sold.")
```
Comment (noliver-ykc) [inline backend/handler/handler.go:385]: Im not sure what this error message means, do you mind clarifying? (you dont have to update the code, just responding to my comment is fine)

File: backend/handler/handler.go [hunk 3]
```diff
@@ -568,9 +697,52 @@ func mapErrorMessage(e validator.FieldError) string {
...
+		ErrMsg = ("The password needs to be 6-20 characters long, and contain at least two groups of the following: uppercase letters, lowercase letters, numbers, and symbols")
 	default:
 		ErrMsg = fmt.Sprintf("%s is not valid", e.Field())
 	}
 
 	return ErrMsg
 }
+
+// Customize validator function for password
+func passwordValidator(fl validator.FieldLevel) bool {
```
Comment (noliver-ykc) [inline backend/handler/handler.go:728]: Thanks for adding all the vaidations!! This looks great

**Review by noliver-ykc — CHANGES_REQUESTED** [review]
Looks good overall, left a few comments here and there. The main thing that should probably be changed is line 318 for allowing .png and .jpg file types. よろしくお願いします

File: backend/handler/handler.go (hunk 2 above)
Comment (xu-jiach) [reply backend/handler/handler.go:385]: the add-items and sell are two separate functions. successfully adding items into the db doesn't mean it is on sale. Therefore, when selling an item, we want to make sure only the item that is pending to be sold can move forward.

File: backend/handler/handler.go (hunk 1 above)
Comment (xu-jiach) [reply backend/handler/handler.go]: "image/jpeg" will accept both jpeg and jpg.
updated the code and accept png right now

File: backend/handler/handler.go (hunk 2 above)
Comment (noliver-ykc) [reply backend/handler/handler.go:385]: I see, so essentially this error is for when someone tries to sell an item that has already been listed?

**Pull request merged**